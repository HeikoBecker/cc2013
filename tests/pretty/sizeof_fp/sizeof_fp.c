int main(int *x) {
&x = sizeof(void (*)(int)); // get the size of a pointer to a int->void function
&x = sizeof(void (*)(int, char)); // get the size of a pointer to a (int, char)->void function
&x = sizeof(x+x); // expression behind sizeof, but () looks like it may contain a type name
}
