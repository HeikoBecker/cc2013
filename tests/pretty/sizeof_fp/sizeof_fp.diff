1,6c1,4
< int (main(int (*x)))
< {
< 	((&x) = (sizeof(void((*)(int)))));
< 	((&x) = (sizeof(void((*)(int, char)))));
< 	((&x) = (sizeof (x + x)));
< }
---
> sizeof_fp.c:2:1: error: The left operand of an assignment must be a lvalue
> &x = sizeof(void (*)(int)); // get the size of a pointer to a int->void function
> ^^^^
> 1 error(s)
