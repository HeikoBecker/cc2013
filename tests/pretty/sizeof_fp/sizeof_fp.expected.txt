int (main(int (*x)))
{
	((&x) = (sizeof(void((*)(int)))));
	((&x) = (sizeof(void((*)(int, char)))));
	((&x) = (sizeof (x + x)));
}
